pkg/operator/csi/csidrivercontrollerservicecontroller/csi_driver_controller_service_controller.go

```diff
@@ -212,30 +212,44 @@ func (c *CSIDriverControllerServiceController) sync(ctx context.Context, syncCon
 
 	_, _, err = v1helpers.UpdateStatus(
 		c.operatorClient,
-		updateStatusFn,
 		v1helpers.UpdateConditionFn(availableCondition),
 		v1helpers.UpdateConditionFn(progressingCondition),
+		updateStatusFn,
 	)
 
 	return err
 }
 
-func isProgressing(status *opv1.OperatorStatus, deployment *appsv1.Deployment) (bool, string) {
+func (c *CSIDriverControllerServiceController) isProgressing(status *opv1.OperatorStatus, deployment *appsv1.Deployment) (bool, string) {
+	if deployment.Generation != deployment.Status.ObservedGeneration {
+		return true, "Waiting for Deployment to act on changes"
+	}
+
 	var deploymentExpectedReplicas int32
 	if deployment.Spec.Replicas != nil {
 		deploymentExpectedReplicas = *deployment.Spec.Replicas
 	}
 
-	switch {
-	case deployment.Generation != deployment.Status.ObservedGeneration:
-		return true, "Waiting for Deployment to act on changes"
-	case deployment.Status.UnavailableReplicas > 0:
-		return true, "Waiting for Deployment to deploy pods"
-	case deployment.Status.UpdatedReplicas < deploymentExpectedReplicas:
-		return true, "Waiting for Deployment to update pods"
-	case deployment.Status.AvailableReplicas < deploymentExpectedReplicas:
-		return true, "Waiting for Deployment to deploy pods"
+	if deployment.Status.AvailableReplicas < deploymentExpectedReplicas {
+		// Verify if the current deployment generation is the same as the one that this controller
+		// dealt with in the previous sync. If they are not the same, we're definitely rolling
+		// out new config, so we are progressing.
+		lastGeneration := resourcemerge.ExpectedDeploymentGeneration(deployment, status.Generations)
+		if deployment.Generation != lastGeneration {
+			return true, "Waiting for Deployment to deploy node pods"
```

Comment:
How can I make these messages more distinct?

Reply:
IMO the message is OK.